Cython/Compiler/Nodes.py

```diff
@@ -8729,11 +8734,7 @@ def cleanup_temps(self, code):
         if self.is_parallel and not self.is_nested_prange:
             code.putln("/* Clean up any temporaries */")
             for temp, type in sorted(self.temps):
-                if type.is_memoryviewslice:
-                    code.put_xdecref_memoryviewslice(temp, have_gil=False)
-                elif type.is_pyobject:
-                    code.put_xdecref(temp, type)
-                    code.putln("%s = NULL;" % temp)
+                code.put_xdecref_clear(temp, type, do_for_memoryviewslice=True, have_gil=False)
```

Comment:
There is a slight difference here. `put_xdecref_clear()` generates code to first clear the variable and then decref it. I think that should often be ok, since the C compiler should generally be able to optimise away the strict ordering for local variables. But it's something that has been bothering me for a while since this difference isn't always explicit in the generator code. Not to be solved right now, but to keep in mind.